common/models/user.js

```diff
@@ -668,6 +668,45 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    var oldEmail, newEmail, oldId, newId;
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        if (ctx.instance.id !== undefined) {
+          ctx.hookState.id = ctx.instance.id;
+          oldId = ctx.hookState.id;
+        }
+      }
+      if (ctx.data) {
+        ctx.hookState.id = ctx.where.id;
+        oldId = ctx.hookState.id;
+      }
+      ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+        userInstances.forEach(function(userInstance) {
+          ctx.hookState.email = userInstance.email;
```

Comment:
Since `userInstances` may have any number of items - zero, one or more - you should store a list of affected users, not only a single instance. For each item, store both `id` and `email`.

```js
ctx.hookState.originalUserData = userInstances.map(function(u) {
  return { id: u.id, email: u.email };
});
```